### PR TITLE
upgrade to rxjava2

### PR DESCRIPTION
**Pull request by venkateshreddy74**
updated to rxjava version to 2

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -31,9 +31,12 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     testCompile 'junit:junit:4.12'
     compile 'com.android.support:appcompat-v7:23.4.0'
-    // RxJava
-    compile 'io.reactivex:rxjava:1.1.5'
-    compile 'io.reactivex:rxandroid:1.2.0'
+    //RxAndroid
+    compile 'io.reactivex.rxjava2:rxandroid:2.0.1'
+// Because RxAndroid releases are few and far between, it is recommended you also
```
Comment (florina-muntenescu) [inline app/build.gradle]: format the file to have the comments start with some spaces, like the "compile..." lines

File: app/src/main/java/upday/droidconmvvm/MainActivity.java [hunk 2]
```diff
@@ -11,16 +11,14 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.disposables.CompositeDisposable;
 import upday.droidconmvvm.model.Language;
 
 public class MainActivity extends AppCompatActivity {
 
-    @NonNull
-    private CompositeSubscription mSubscription;
+    @NonNull CompositeDisposable compositeDisposable;
```
Comment (florina-muntenescu) [inline app/src/main/java/upday/droidconmvvm/MainActivity.java]: make this private, keep the `@NonNull` on a different line, name this `mCompositeDisposable`

File: app/src/main/java/upday/droidconmvvm/datamodel/DataModel.java [hunk 3]
```diff
@@ -5,7 +5,8 @@
 import java.util.Arrays;
 import java.util.List;
 
-import rx.Observable;
+//import rx.Observable;
```
Comment (florina-muntenescu) [inline app/src/main/java/upday/droidconmvvm/datamodel/DataModel.java]: remove the commented line

File: app/src/main/java/upday/droidconmvvm/schedulers/ISchedulerProvider.java [hunk 4]
```diff
@@ -2,7 +2,9 @@
 
 import android.support.annotation.NonNull;
 
-import rx.Scheduler;
+import io.reactivex.Scheduler;
+
+//import rx.Scheduler;
```
Comment (florina-muntenescu) [inline app/src/main/java/upday/droidconmvvm/schedulers/ISchedulerProvider.java]: remove the commented line

File: app/src/main/java/upday/droidconmvvm/schedulers/SchedulerProvider.java [hunk 5]
```diff
@@ -3,9 +3,13 @@
...
 import android.support.annotation.Nullable;
 
-import rx.Scheduler;
-import rx.android.schedulers.AndroidSchedulers;
-import rx.schedulers.Schedulers;
+import io.reactivex.Scheduler;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+//import rx.Scheduler;
```
Comment (florina-muntenescu) [inline app/src/main/java/upday/droidconmvvm/schedulers/SchedulerProvider.java]: remove the commented lines

File: app/src/main/java/upday/droidconmvvm/MainActivity.java (hunk 2 above)
Comment (venkateshreddy74) [reply app/src/main/java/upday/droidconmvvm/MainActivity.java]: done

File: app/build.gradle (hunk 1 above)
Comment (venkateshreddy74) [reply app/build.gradle]: done

File: app/src/main/java/upday/droidconmvvm/datamodel/DataModel.java (hunk 3 above)
Comment (venkateshreddy74) [reply app/src/main/java/upday/droidconmvvm/datamodel/DataModel.java]: done

File: app/src/main/java/upday/droidconmvvm/schedulers/ISchedulerProvider.java (hunk 4 above)
Comment (venkateshreddy74) [reply app/src/main/java/upday/droidconmvvm/schedulers/ISchedulerProvider.java]: done

File: app/src/main/java/upday/droidconmvvm/schedulers/SchedulerProvider.java (hunk 5 above)
Comment (venkateshreddy74) [reply app/src/main/java/upday/droidconmvvm/schedulers/SchedulerProvider.java]: done

**Comment by florina-muntenescu** [comment]
Thanks for the PR!

**Comment by venkateshreddy74** [comment]
My Pleasure !

Thanks for the quick review :)


Thanks
Venkatesh Reddy
GRA
CSSE
________________________________

**Pull request merged**